### PR TITLE
Fixes #15016: Merge API documentation for version 2-8

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15016

### CODE REVIEW

File: src/api_v2/datasources.md [hunk 1]
```diff
@@ -363,7 +363,7 @@ curl -H "X-API-Token: yourToken -X PUT http://rudder.example.com/rudder/api/late
 
   /**
   @api {post} /api/datasources/reload 6. Update properties from data source - all nodes
-  @apiVersion 4.0.0
+  @apiVersion 9.0.0
```
Comment (amousset) [inline src/api_v2/datasources.md]: it is not in the version 8?

Comment (fanf) [reply src/api_v2/datasources.md]: not, it was an error previously

Comment (fanf) [reply src/api_v2/datasources.md]: (datasources didn't existed at all before that, and actually, it's a plugin, so I'm not even sure it should be their, or we miss a lot of other plugins, but that discussion belongs elsewhere)

**Comment by amousset** [comment]
Could you also edit the list of version API to mention <8 have been removed from the doc?

**Comment by fanf** [comment]
Commit modified

File: src/introduction.md [hunk 2]
```diff
@@ -46,16 +47,16 @@ curl -X GET -H <span class="str">"X-API-Token: yourToken"</span> http://rudder.e
...
+curl -X GET -H <span class="str">"X-API-Token: yourToken"</span> -H <span class="str">"X-API-Version: <span class="kwd">10</span>" http://rudder.example.com/rudder/api/rules</span>
 <span class="com"># Wrong => Error response indicating which versions are available</span>
 curl -X GET -H <span class="str">"X-API-Token: yourToken"</span> -H <span class="str">"X-API-Version: <span class="kwd">3.14</span>" http://rudder.example.com/rudder/api/rules</span>
 </code></pre>
 
 In the future, we may declare some versions as deprecated, in order to remove them in a later version of Rudder, but we will never remove any versions without warning, or without a safe
 period of time to allow migration from previous versions.
 
-<h4>Existing versions</h4>
+<h4>Existing versions (pre- Rudder v4</h4>
```
Comment (amousset) [inline src/introduction.md]: ?

**Comment by fanf** [comment]
Commit modified

**Pull request merged**